test/elm/interval/test.coffee

```diff
@@ -1405,3 +1405,699 @@ describe 'Collapse', ->
+    # define ClosedSinglePer1: expand { Interval[2, 4] } per 1 '1'
+    a = @closedSinglePer1.exec(@ctx)
+    prettyList(a).should.equal "{ [2, 2], [3, 3], [4, 4] }"
+    
+    # define ClosedSinglePer3: expand { Interval[2, 10] } per 3 '1'
+    a = @closedSinglePer3.exec(@ctx)
+    prettyList(a).should.equal "{ [2, 4], [5, 7], [8, 10] }"
+    
+    # define ClosedSinglePer3Weird: expand { Interval[2, 4] } per 3 '1'
+    a = @closedSinglePer3NoteTheWidth.exec(@ctx)
+    prettyList(a).should.equal "{ [2, 4] }"
+    
+  it 'expands lists of multiple intervals', ->
+    # define NullInList: expand { Interval[2, 4], null } per 1 '1'
+    a = @nullInList.exec(@ctx)
+    prettyList(a).should.equal "{ [2, 2], [3, 3], [4, 4] }"
+
+    # define Overlapping: expand { Interval[2, 4], Interval[3, 5] } per 1 '1'
+    a = @overlapping.exec(@ctx)
+    prettyList(a).should.equal "{ [2, 2], [3, 3], [4, 4], [5, 5] }"
+
+    # define NonOverlapping: expand { Interval[2, 4], Interval[6, 6] } per 1 '1'
+    a = @nonOverlapping.exec(@ctx)
+    prettyList(a).should.equal "{ [2, 2], [3, 3], [4, 4], [6, 6] }"
+
+  it 'expands interval using default per of 1', ->
+    # define NoPer: expand { Interval[2, 4] }
+    a = @noPer.exec(@ctx)
+    prettyList(a).should.equal "{ [2, 2], [3, 3], [4, 4] }"
+
+  it 'expands interval with open ends', ->
+    # define OpenStart: expand { Interval(2, 4] } per 1 '1'
+    a = @openStart.exec(@ctx)
+    prettyList(a).should.equal "{ [3, 3], [4, 4] }"
+
+    # define OpenEnd: expand { Interval[2, 4) } per 1 '1'
+    a = @openEnd.exec(@ctx)
+    prettyList(a).should.equal "{ [2, 2], [3, 3] }"
+
+    # define OpenBoth: expand { Interval(2, 4) } per 1 '1'
+    a = @openBoth.exec(@ctx)
+    prettyList(a).should.equal "{ [3, 3] }"
+
+  it 'returns an empty list if we get an empty list or if there are no results', ->
+    # define EmptyList: List<Interval<Integer>>{}
+    a = @emptyList.exec(@ctx)
+    a.length.should.equal 0
+
+    # define PerTooBig: expand { Interval[2, 4], null } per 5 '1'
+    a = @perTooBig.exec(@ctx)
+    a.length.should.equal 0
+
+  it 'returns null with open ended intervals', ->
+    # define NullClose: expand { Interval[2, null] } per 1 '1'
+    a = @nullClose.exec(@ctx)
+    should.not.exist(a)
+    # define NullOpen: expand { Interval[null, 4] } per 1 '1'
+    a = @nullOpen.exec(@ctx)
+    should.not.exist(a)
+    # define NullBoth: expand { Interval[null, null] } per 1 '1'
+    a = @nullBoth.exec(@ctx)
+    should.not.exist(a)
+
+  it 'returns null when per not applicable or mistmach interval', ->
+    # define BadPerMinute: expand { Interval(2, 4] } per 1 minute
+    a = @badPerMinute.exec(@ctx)
+    should.not.exist(a)
+
+describe 'DecimalIntervalExpand', ->
+  @beforeEach ->
+    setup @, data
+
+  it 'expands single intervals', ->
+    # define ClosedSingle: expand { Interval[2, 5] } per 1.5 '1'
+    a = @closedSingle.exec(@ctx)
+    prettyList(a).should.equal "{ [2, 3.49999999], [3.5, 4.99999999] }"
```

Comment:
Again, open interval endpoints would be neater if possible (but it is spec-compliant as-is)
```
{ [2, 3.5) [3.5, 5.0) }
```